lib/Models/WebMapServiceCatalogItem.js

```diff
@@ -427,10 +427,43 @@ WebMapServiceCatalogItem.prototype.updateFromCapabilities = function(capabilitie
 WebMapServiceCatalogItem.prototype._load = function() {
     if (!defined(this._rawMetadata)) {
         var that = this;
-        return loadXML(proxyCatalogItemUrl(this, this.metadataUrl)).then(function(xml) {
+		
+        var p1 = loadXML(proxyCatalogItemUrl(this, this.metadataUrl)).then(function(xml) {
             var metadata = capabilitiesXmlToJson(xml);
             that.updateFromCapabilities(metadata, false);
         });
+		
+		// Query WMS for wfs or wcs URL if no dataUrl is present
+		if (!defined(this.dataUrl)) {
```

Comment:
We should do this even if `!defined(this._rawMetadata)` (above) is false.  For example, click Add Data in the UI and enter this URL:
http://data.gov.au/geoserver/ballarat-aquatic-facilities/wms?service=WMS&version=1.3.0&request=GetCapabilities

Then, click the Info button next to Ballarat Aquatic Facilities.  The WFS DataURL doesn't appear because DescribeLayer is not invoked because the catalog item already has metadata (provided by its group).

I'm not sure if I've explained that well, so let me know if it's unclear.